### PR TITLE
Update to Polymer 2

### LINKED ISSUE #71: Broken history/scroll support
Despite the URL is properly updated with the correct hash when navigating backward/forward the history, the scroll remains at the top of the page. Not sure if this worked before.

Chrome 54.0.2840.98 on macOS.

### PR DESCRIPTION
**Pull request by TimvdLippe**
This PR updates this website to use Polymer 2. The primary change was the upgrade from paper-* to app-* as the original paper elements were deprecated. Other than that, as a result of Polymer hybrid, there are no behavioral changes. The PR includes two commits, to separate the bower_components noise from the actual work. For reviewing, I advise to only look at the second commit.

I did a visual comparison between the live version and this version and if I am correct all styling should have been updated accordingly.

Fixes #71

### CODE REVIEW

**Comment by kdzwinel** [comment]
😲 that's AMAZING, awesome job 👏 

No way I'm going to read 150k lines of code though 😉 Can you please make it available to play with via github pages?

**Comment by paulirish** [comment]
whoa

**Comment by TimvdLippe** [comment]
@kdzwinel the actual upgrade is included in the second commit and it should only be ~200 deletions and ~200 additions. The first commit is simply the result of updating `bower.json` to target all dependencies to `2.0.0` and run `bower install` freshly.

I can try to get a running example available if that makes reviewing easier.

**Comment by TimvdLippe** [comment]
https://timvdlippe.github.io/devtools-protocol/ has a staging environment.

```js
> Polymer.version
< "2.1.0"
```

**Comment by kdzwinel** [comment]
> the actual upgrade is included in the second commit 

yeah, I missed that at first glance 👍 

Some bugs I've found playing with staging:
- search isn't working
- version picker is a bit narrower than it should be
- search results box is visible at the bottom of the page

screen shot 2017-09-25 at 22 58 01

The really good news is that, as far as I can see, your PR fixes #71 !

File: _layouts/default.html [hunk 1]
```diff
@@ -7,48 +7,54 @@
...
-  <link rel="import" href="{{ site.baseurl }}/bower_components/paper-drawer-panel/paper-drawer-panel.html">
   <link rel="import" href="{{ site.baseurl }}/bower_components/iron-icons/iron-icons.html">
+  <link rel="import" href="{{ site.baseurl }}/bower_components/iron-selector/iron-selector.html">
 
   <link rel="import" href="{{ site.baseurl }}/elements/cr-search-control/cr-search-control.html">
 
   <link href="{{ site.baseurl }}/styles/protocol.css" rel="stylesheet">
   <link rel="import" href="{{ site.baseurl }}/styles/app-theme.html">
 </head>
-<body unresolved class="fullbleed layout vertical">
```
Comment (kdzwinel) [inline _layouts/default.html]: I think we still need sth like `unresolved` to hide content before it's loaded. There is a really bad FOUC on every page load in Firefox and Safari.

fouc mov

**Comment by TimvdLippe** [comment]
Most of the styling issues are fixed, but the `<iron-list>` is missing search results. I couldn't figure this out at a glance, but I have to work on something else for now. Will get back to the search results after that.

**Comment by TimvdLippe** [comment]
Oh and staging has the updates as well so you can test it again :smile:

**Comment by TimvdLippe** [comment]
Staging should now work properly again with the search results and input. Please check it out :smile:

**Comment by paulirish** [comment]
@TimvdLippe are you open to flattening the imports? we've noticed the perf of this app is pretty rough on the network. we could reasonably do this in a build step before pushing to the gh-pages branch.

**Comment by TimvdLippe** [comment]
@paulirish Oh yes sure. It is probably a good idea to do that in a separate PR though. We can integrate the Polymer CLI or Polymer build to do that for us.

**Comment by kdzwinel** [comment]
@TimvdLippe I still see a lot of weirdness in safari 11. Are we missing some kind of polyfill?

not styled input element:
screen shot 2017-09-27 at 09 53 51

header missing styling after page reload from time to time:
screen shot 2017-09-27 at 09 54 17

search results taking whole screen (covering input) on narrow screens:
screen shot 2017-09-27 at 09 57 12

**Comment by TimvdLippe** [comment]
Hm, I will need to dig deeper into that tomorrow. After friday I wont have access to a Safari browser, so I hope I can fix it before that time. Do the artifacts also show up in other non-Chrome browsers?

**Comment by TimvdLippe** [comment]
I have been debugging a lot and was able to fix the header. The `<input type="search">` is apparently breaking. It does have the correct styling, but then it just discards that styling when actually rendering the element. Changing it to `<input type="text">` fixes this issue.

The weird thing about this artifact is that it was working with Polymer 1, that's what I do not understand. I have to do more debugging and researching on styling this input element to hopefully resolve this issue.

**Comment by TimvdLippe** [comment]
Okay, with some help of @usergenic we were able to figure it out. The `-webkit-appearance: textfield` style on the input element fixed most of the styling. Could you take another look @kdzwinel 

Note that for some reason the website is aggressively cached and it was still serving the old code for me.  We checked on a different Mac with a fresh load and there it was fixed :tada:

**Comment by kdzwinel** [comment]
Styling looks fine now, thanks!

I'm not quite sure what's happening (support of `position:sticky`?), but:
- header is not sticky for me on stable chrome
- while it's sticky on safari and firefox.

🤔

And when header is sticky, there is a bug in the search widget:
- go to any domain,
- scroll down,
- try searching.

Search results will cover the header making search unusable.

**Comment by TimvdLippe** [comment]
@kdzwinel Good catches! Last commit should fix these issues :smile:

**Comment by TimvdLippe** [comment]
Friendly ping, I think this is ready to go. I couldn't find any more issues with search and Safari 😛

**Comment by paulirish** [comment]
Let's ship it then! 
Thanks man!

**Comment by paulirish** [comment]
I did before and after Lighthouse runs on this and the performance is consistently about 5% better. Which is fine for me!

here's the before after of the trace: 

image

basically consistent. seems good.

**Comment by paulirish** [comment]
@TimvdLippe wanna try flattening the imports? check this out:

image

**Comment by TimvdLippe** [comment]
Yeah, we should probably adopt polymer build to optimize the bundle. Cc @fredkschott for thoughts/experiences with Jekyll?

**Pull request merged**